pilot/cmd/pilot-agent/main.go

```diff
@@ -715,6 +717,8 @@ func init() {
 		"Process bootstrap provided via templateFile to be used by control plane components.")
 	proxyCmd.PersistentFlags().StringVar(&outlierLogPath, "outlierLogPath", "",
 		"The log path for outlier detection")
+	proxyCmd.PersistentFlags().StringVar(&gcpConfigPath, "gcpConfigPath", "",
```

Comment:
Can't we not add this new flag?  Do something similar to platform discovery herehttps://github.com/istio/istio/blob/master/pkg/bootstrap/platform/discovery.go#L26-L28

Reply:
I'd like to find a better way to do this as well. The use case is when pilot-agent is running on-premise, where locality and credentials need to be provided through parameters/config files manually. Please let me know if you have a better idea on how to do this.

Reply:
Can you just use https://github.com/istio/istio/tree/master/samples/custom-bootstrap?

Reply:
It would work. But we'd like to make the experience more transparent where a user doesn't have to add the overwrite annotation everywhere.

Reply:
You don't need to use that directly but possible using the same mechanism of using the overrides. They already have to pass a `GCPConfigPath` file as a flag, why not pass some overlay for the config that applies those things?

Reply:
I was planning to modify the sidecar injector to pass GCPConfigPath. If sidecar injector also has the power to add the annotation, it should work for us.